src/account_manager.rs

```diff
@@ -171,22 +240,24 @@ impl AccountManager {
         from_account_id: &AccountIdentifier,
         to_account_id: &AccountIdentifier,
         amount: u64,
-    ) -> crate::Result<InternalTransferMetadata> {
-        let mut from_account = self.get_account(from_account_id)?;
-        let to_account = self.get_account(to_account_id)?;
-        let to_address = to_account
-            .latest_address()
-            .ok_or_else(|| anyhow::anyhow!("destination account address list empty"))?
-            .clone();
-        let from_synchronized = from_account.sync().execute().await?;
-        let metadata = from_synchronized
+    ) -> crate::Result<Message> {
+        let from_account_guard = self.get_account(from_account_id)?;
+        let to_account_guard = self.get_account(to_account_id)?;
+
+        let to_address = {
+            let to_account = to_account_guard.read().unwrap();
+            to_account
+                .latest_address()
+                .ok_or_else(|| anyhow::anyhow!("destination account address list empty"))?
```

Comment:
Out of interest, is it necessary to use `anyhow`? The crate dynamically allocates space for errors. It's often more performant or even easier to just use a enum. This also means that any crates built on top of this API can handle different errors in different ways instead of needing to inspect and opaque error type.

Reply:
yeah it's on my todo list and I think i'll do it today